pkg/dao/rpm_test.go

```diff
@@ -263,3 +355,109 @@ func (s *RpmSuite) TestRpmSearchError() {
 	assert.Equal(t, err.Error(), "request.URLs must contain at least 1 URL")
 	tx.RollbackTo(txSP)
 }
+
+type TestInsertForRepositoryCase struct {
+	given    int
+	expected string
+}
+
+var testCases []TestInsertForRepositoryCase = []TestInsertForRepositoryCase{
+	{
+		given:    scenario0,
+		expected: "empty slice found",
+	},
+	{
+		given:    scenario3,
+		expected: "",
+	},
+	{
+		given:    scenarioUnderThreshold,
+		expected: "",
+	},
+	{
+		given:    scenarioThreshold,
+		expected: "",
+	},
+	{
+		given:    scenarioOverThreshold,
+		expected: "",
+	},
+}
+
+func (s *RpmSuite) genericInsertForRepository(testCase TestInsertForRepositoryCase) {
+	t := s.Suite.T()
+	tx := s.tx
+
+	pagedRpmInsertsLimit := 100
+	dao := GetRpmDao(tx, map[string]interface{}{
+		OptionPagedRpmInsertsLimit: pagedRpmInsertsLimit,
+	})
+
+	p := s.prepareScenarioRpms(testCase.given, pagedRpmInsertsLimit)
+	records, err := dao.InsertForRepository(s.repo.Base.UUID, p)
+
+	if testCase.expected != "" {
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), testCase.expected)
+	} else {
+		assert.NoError(t, err)
+		assert.Equal(t, int64(len(p)), records)
+	}
```

Comment:
It might be good to also actually query to make sure that the packages are associated to the repository, or at least the count is what you'd expect.